chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/urluptime/UrlUptimeCalculator.java

```diff
@@ -0,0 +1,85 @@
+import org.springframework.stereotype.Component;
+
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2(topic = "urlUptimeEmailJobLogger")
+@Component
+public class UrlUptimeCalculator {
+
+    private ChplUptimeMonitorDAO chplUptimeMonitorDAO;
+    private ChplUptimeMonitorTestDAO chplUptimeMonitorTestDAO;
+
+    @Autowired
+    public UrlUptimeCalculator(ChplUptimeMonitorDAO chplUptimeMonitorDAO, ChplUptimeMonitorTestDAO chplUptimeMonitorTestDAO) {
+        this.chplUptimeMonitorDAO = chplUptimeMonitorDAO;
+        this.chplUptimeMonitorTestDAO = chplUptimeMonitorTestDAO;
+    }
+
+    public List<UrlUptimeReport> calculateRowsForReport() {
+        List<UrlUptimeReport> reports = new ArrayList<UrlUptimeReport>();
+
+        getChplUptimeMonitors().stream()
+                .forEach(monitor -> reports.add(summarize(monitor, getChplUptimeMonitorTests(monitor.getId()))));
+        return reports;
+    }
+
+    private List<ChplUptimeMonitor> getChplUptimeMonitors() {
+        return chplUptimeMonitorDAO.getAll();
+    }
+
+    private List<ChplUptimeMonitorTest> getChplUptimeMonitorTests(Long chplUptimeMonitorId) {
+        return chplUptimeMonitorTestDAO.getChplUptimeMonitorTests(chplUptimeMonitorId);
+    }
+
+    private UrlUptimeReport summarize(ChplUptimeMonitor chplUptimeMonitor, List<ChplUptimeMonitorTest> chplUptimeMonitorTests) {
+        LOGGER.info("Summarizing data for {}", chplUptimeMonitor.getUrl());
+
+        List<ChplUptimeMonitorTest> allTestsForPastWeek = getEligibleTestsForPastWeek(chplUptimeMonitorTests);
+        List<ChplUptimeMonitorTest> allTestsForCurrentMonth = getEligibleTestsForCurrentMonth(chplUptimeMonitorTests);
+
+        return UrlUptimeReport.builder()
+                .description(chplUptimeMonitor.getDescription())
+                .url(chplUptimeMonitor.getUrl())
+                .totalTestCount(Long.valueOf(chplUptimeMonitorTests.size()))
+                .totalSuccessfulTestCount(chplUptimeMonitorTests.stream()
+                        .filter(test -> test.getPassed())
+                        .count())
+                .currentMonthTestCount(Long.valueOf(allTestsForCurrentMonth.size()))
+                .currentMonthSuccessfulTestCount(allTestsForCurrentMonth.stream()
+                        .filter(test -> test.getPassed())
+                        .count())
+                .pastWeekTestCount(Long.valueOf(allTestsForPastWeek.size()))
+                .pastWeekSuccessfulTestCount(allTestsForPastWeek.stream()
+                        .filter(test -> test.getPassed())
+                        .count())
+                .build();
+    }
+
+    private List<ChplUptimeMonitorTest> getEligibleTestsForPastWeek(List<ChplUptimeMonitorTest> chplUptimeMonitorTests) {
+        LocalDateTime end = LocalDateTime.now().minusDays(1).with(ChronoField.NANO_OF_DAY, LocalTime.MAX.toNanoOfDay());
+        LocalDateTime start = LocalDateTime.now().minusDays(6).with(ChronoField.NANO_OF_DAY, LocalTime.MIN.toNanoOfDay());
```

Comment:
Should this be?

```suggestion
        LocalDateTime start = LocalDateTime.now().minusDays(7).with(ChronoField.NANO_OF_DAY, LocalTime.MIN.toNanoOfDay());
```

I can see that you're going to the end of the day yesterday on line 68, and the beginning of the day on line 69, but I think it needs to go back one day further to have a full seven days of data